modules/apps/29-fee/types/msgs.go

```diff
@@ -171,8 +144,60 @@ func NewIdentifiedPacketFee(packetId *channeltypes.PacketId, fee Fee, refundAddr
 	}
 }
 
+func (fee IdentifiedPacketFee) Validate() error {
+	if err := host.PortIdentifierValidator(fee.PacketId.PortId); err != nil {
+		return sdkerrors.Wrap(err, "invalid source port ID")
+	}
+
+	if err := host.ChannelIdentifierValidator(fee.PacketId.ChannelId); err != nil {
+		return sdkerrors.Wrap(err, "invalid source channel ID")
+	}
+
+	if fee.PacketId.Sequence == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidSequence, "packet sequence cannot be 0")
+	}
+
+	_, err := sdk.AccAddressFromBech32(fee.RefundAddress)
+	if err != nil {
+		return sdkerrors.Wrap(err, "failed to convert RefundAddress into sdk.AccAddress")
+	}
+
+	// if any of the fee's are invalid return an error
+	if !fee.Fee.AckFee.IsValid() || !fee.Fee.ReceiveFee.IsValid() || !fee.Fee.TimeoutFee.IsValid() {
+		return sdkerrors.ErrInvalidCoins
+	}
+
+	// if all three fee's are zero or empty return an error
+	if fee.Fee.AckFee.IsZero() && fee.Fee.ReceiveFee.IsZero() && fee.Fee.TimeoutFee.IsZero() {
+		return sdkerrors.ErrInvalidCoins
+	}
+
+	// enforce relayer is nil
+	if fee.Relayers != nil {
+		return ErrRelayersNotNil
+	}
+
+	return nil
+}
+
 // NewPacketId returns a new instance of PacketId
 // TODO: move to channeltypes
 func NewPacketId(channelId, portId string, seq uint64) *channeltypes.PacketId {
 	return &channeltypes.PacketId{ChannelId: channelId, PortId: portId, Sequence: seq}
 }
+
+// Validates a PacketId
+// TODO: move to channeltypes
+/*
+func (p channeltypes.PacketId) Validate() error {
```

Comment:
leaving in as we can copy and paste to channeltypes PR later.

Reply:
ahh i moved it in #551, you can move it as well. Or just make it a non-object method for now (i.e. ValidatePacketId(pid PacketId) and move it later .

Or just do the same change and resolve merge conflict later

Reply:
Ok moved it over and I'm using the packetId validation now. Sorry about the confusion there lol.